internal/clients/agora.go

```diff
@@ -0,0 +1,133 @@
+package clients
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/tuannguyenandpadcojp/duck-bot/pkg/utils"
+)
+
+type AgoraConfig struct {
+	BaseURL        string
+	AppID          string
+	CustomerID     string
+	CustomerSecret string
+}
+
+type Agora struct {
+	c   *http.Client
+	cfg AgoraConfig
+}
+
+type UsageRequest struct {
+	FromDate string `json:"from_date"` //2024-01-01
+	ToDate   string `json:"to_date"`   //2024-01-05
+}
+
+type Usage struct {
+	DurationAudioAll   int `json:"durationAudioAll"`
+	DurationVideo1080P int `json:"durationVideo1080P"`
+	DurationVideo2K    int `json:"durationVideo2K"`
+	DurationVideo4K    int `json:"durationVideo4K"`
+	DurationVideoHd    int `json:"durationVideoHd"`
+	DurationVideoHdp   int `json:"durationVideoHdp"`
+}
+
+type UsageResponse struct {
+	Date  time.Time `json:"date"`
+	Usage Usage     `json:"usage"`
+}
+
+type UsagesResponse struct {
+	Usages []UsageResponse `json:"usages"`
+}
+
+func NewAgora(cfg AgoraConfig) *Agora {
+	return &Agora{
+		c:   utils.NewHTTPClient(),
+		cfg: cfg,
+	}
+}
+
+func (a *Agora) GetUsage(ctx context.Context, req UsageRequest) (*UsagesResponse, error) {
+	endpoint := fmt.Sprintf("%s%s", a.cfg.BaseURL, "/v3/usage")
+	request, err := http.NewRequest(http.MethodGet, endpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	params := url.Values{}
+	params.Set("project_id", a.cfg.AppID)
+	params.Set("business", "default")
+	params.Set("from_date", req.FromDate)
+	params.Set("to_date", req.ToDate)
+
+	request.URL.RawQuery = params.Encode()
+	request.Header.Set("Authorization", fmt.Sprintf("Basic %s", a.credentials()))
```

Comment:
Using http.Request basicAuth method
```suggestion
	request.SetBasicAuth(a.cfg.CustomerID, a.cfg.CustomerSecret)
```